File: erykandroidcommon/src/androidTest/java/pl/edu/amu/wmi/erykandroidcommon/di/MyDaggerMockRule.java
package pl.edu.amu.wmi.erykandroidcommon.di;

import android.app.Application;
import android.support.test.InstrumentationRegistry;

import it.cosenonjaviste.daggermock.DaggerMockRule;

public class MyDaggerMockRule extends DaggerMockRule<TestComponent> {

    public MyDaggerMockRule() {
        super(TestComponent.class, new TestModule(getApp()));
        set(new ComponentSetter<TestComponent>() {
            @Override
            public void setComponent(TestComponent component) {
//                getApp().setComponent(component);
            }
        });
    }

    private static Application getApp() {
        return (Application) InstrumentationRegistry.getInstrumentation()
                .getTargetContext().getApplicationContext();
    }

}

File: erykandroidcommon/src/androidTest/java/pl/edu/amu/wmi/erykandroidcommon/di/MyDaggerMockRule.kt
package pl.edu.amu.wmi.erykandroidcommon.di

import android.app.Application
import android.support.test.InstrumentationRegistry

import it.cosenonjaviste.daggermock.DaggerMockRule

class MyDaggerMockRule : DaggerMockRule<TestComponent>(TestComponent::class.java, TestModule(app)) {
    
    init {
        set {
            //                getApp().setComponent(component);
        }
    }

    companion object {
        private val app: Application
            get() = InstrumentationRegistry.getInstrumentation()
                    .targetContext.applicationContext as Application
    }

}
